Security login 구현

Flre-fly opened the pull request:
### keword 
- KEYWORD #{22}

### summary
- 로그인, 로그아웃 기능 + test
- admin권한을 가진 user을 등록시키고 등록된 user의 id/pw로 login test
- 회원탈퇴기능(탈퇴 시 관련 정보들을 모두 del처리)

### as-is / to-be
- login 기능x / login 관련 기능 + 테스트

### review point
- 빠진 부분은 없는지 코드 리팩터링이 필요한지를 중심으로 코드리뷰 해주세요

Flre-fly committed:
backend-22 login 구현
-> feat : 회원가입 기능 추가 & test
-> refactoring :  <set>이었던 role을 그냥 enum으로 변경

Flre-fly committed:
backend-22 login 구현
-> refactoring :  기존의 role의 이름을 ADMIN -> ROLE_ADMIN으로 중간에 변환하는 과정을 조금 더 단순하게 함
: AdminService에 있는 내용을 UserDetailService로 옮김

Flre-fly committed:
backend-22 login 구현
-> test : admin과 member를 함께 저장하는 test
-> refactoring : admin의 loginId를 unique=true로 변경

Flre-fly committed:
backend-22 login 구현
-> test : admin과 member를 함께 update, delete하는 test작성

Flre-fly committed:
backend-22 login 구현
-> test : samplecontroller로 계정만들고 로그인한다음에 해당 페이지에 접근가능한것 확인
: /logout페이지로 들어가면 로그아웃되는것 확인
-> feat : /h2-console 접근을 허용하기 위한 코드 추가

Flre-fly committed:
Pull Request 템플릿 생성

Flre-fly committed:
backend-22 login 구현
-> feat : 회원탈퇴기능구현

Flre-fly commented:
수정사항 : admindto가 없는 사람의 member등록 요청 처리

Flre-fly commented:
수정사항 : 같은 id로 회원가입을 시도하다가 실패할때 설정한 예외가 잘 터지는지도 test

Byuntil reviewed (approved):
그 memberService에서 update하는 부분의 반환값을 엔티티로 반환한 이유가 있어?
말그대로 update니까 void여도 될것같은데(어차피 아이디는 기존 member랑 똑같으니)

내가 잘 본건지는 모르겠는데 memberService부분 updateMember에 비밀번호 encoded하는 부분이 있던데
userDetailService에 있는 encodedPw가 또 있는데 memberService부분 updateMember안에 userDetailService에 있는 encodedPw()를 쓰는게 좋을것같은데 

memberService부분 update에 userDetailService에 있는 encodedPw사용해야 할것 같고
이어지는 이야기인데 memberServiceTest 부분에서 멤버업데이트 부분에서 새로운 id 새로운 비번등을 담은 updateAdminDto에 넣고 memberService.updateMember()해서 업데이트가 되는데 이 메서드 안에서도 패스워드 encoded했는데 또 userDetailService에서 또 encodedPw하는 이유가 있어?

Flre-fly committed:
backend-22 login 구현
-> refactoring :
- memberService.update의 return값 void로 설정
- memberService에서 passwordEncoded사용 대신 userDetailService에 있는 encodedPw사용
- 멤버업데이트 test에서 중복으로 encoded되는 부분 수정
